invoice/src/main/java/org/killbill/billing/invoice/InvoiceDispatcher.java

```diff
@@ -651,8 +651,8 @@ public boolean apply(final InvoiceItem input) {
                 commitInvoiceAndSetFutureNotifications(account, null, futureAccountNotifications, internalCallContext);
             }
 
-            if (success) {
-                final DefaultInvoice refreshedInvoice = new DefaultInvoice(invoiceDao.getById(invoice.getId(), internalCallContext));
+            if (isDryRun || success) {
```

Comment:
Seems weird to call `onSuccessCall` if `isDryRun` its `true` but `success` is `false` ?

Reply:
Indeed, but it's because how the code is written: if `isDryRun`, `success` is always false (`success` means that we could write to disk the invoice and notifications if not dry run).